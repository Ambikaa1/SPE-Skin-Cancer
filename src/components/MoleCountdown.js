import React from "react";
import { View, Text, StyleSheet } from "react-native";

const MoleCountdown = ({ item, index }) => {
    const now = Date.now()
    const difference = item.nextUpdate - now
    const percentage = difference / 2592000000;

    const barStyle = () => {
        console.log(percentage);
        return ({
            height: "100%",
            width: `${percentage * 100}%`,
            backgroundColor: "#71A1D1"
        })
    }

    const numOfDays = () => {
        const days = Math.floor(difference / 86400000)
        let message;
        if (days <= 0) {
            message = "Today"
        } else {
            message = days + " days"
        }
        return message
    }

    return (
        <View style = {(index === 0) ? styles.containerTop : styles.container}>
            <View style = {styles.info}>
                <Text style = {styles.name} numberOfLines = {1}>{item.name}</Text>
                <Text style = {styles.numDays}>{numOfDays()}</Text>
            </View>
            <View style = {styles.barContainer}>
                <View style = {styles.bar}>
                    <View style = {barStyle()} />
                    <View style = {styles.remainingBar} />
                </View>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    containerTop: {
        paddingVertical: 10,
        borderTopWidth: 0.5,
        borderBottomWidth: 0.5,
        borderColor: "#AAAAAA",
    },
    container: {
        paddingVertical: 10,
        borderBottomWidth: 0.5,
        borderColor: "#AAAAAA",
    },
    info: {
        flexDirection: "row",
        justifyContent: "space-between"
    },
    name: {
        fontSize: 17,
        flex: 1
    },
    numDays: {
        fontSize: 17
    },
    barContainer: {
        marginTop: 5,
    },
    bar: {
        flexDirection: "row",
        height: 10,
        borderRadius: 5,
        overflow: "hidden"
    },
    remainingBar: {
        height: "100%",
        backgroundColor: "#71A1D120",
        opacity: 10,
        flex: 1,
    }
});

export default MoleCountdown;