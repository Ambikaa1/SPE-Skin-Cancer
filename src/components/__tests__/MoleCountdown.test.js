import React from 'react';
import renderer from 'react-test-renderer';

import MoleCountdown from "../MoleCountdown";

describe('<MoleCountdown />', () => {
    mockItem = {nextUpdate: 0}
    it('renders correctly', () => {
        const tree = renderer.create(<MoleCountdown item = {mockItem} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
