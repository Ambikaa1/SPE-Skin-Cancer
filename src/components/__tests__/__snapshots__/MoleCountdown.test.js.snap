// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoleCountdown /> renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0.5,
      "borderColor": "#AAAAAA",
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "flex": 1,
          "fontSize": 17,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      Today
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 5,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 5,
          "flexDirection": "row",
          "height": 10,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#71A1D1",
            "height": "100%",
            "width": "-62431.75854301697%",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#71A1D120",
            "flex": 1,
            "height": "100%",
            "opacity": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
